actionRules - AlertsManagement

roramc opened the pull request:
adding new resource type actionRules in AlertsManagement RP

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

roramcmsft committed:
actionRules - AlertsManagement

adding new resource type actionRules in AlertsManagement RP

azuresdkci commented:
Hello!!

The Rest API Specs team wishes everyone a happy holiday and would like to kindly inform you that responses and review to Pull request will be delayed during the holiday period (now -> 2nds of January) to allow for Awesome reviewers to have an awesome time with their families during the holidays!

Thanks and Have a WONDERFUL HOLIDAY

AutorestCI commented:
# Automation for azure-sdk-for-js

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-python

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-go

nschonni commented on specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/examples/ActionRules_Delete.json:
```diff
@@ -0,0 +1,15 @@
+	"responses": {
+		"200": {
+			"body": {
+				true	
```
This wouldn't be valid JSON

roramc replied:
resolved in latest commit

roramc replied:
resolved in latest commit

nschonni commented on specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json:
```diff
@@ -0,0 +1,2062 @@
+            "Monthly"
+          ],
+          "x-ms-enum": {
+            "name": "SupressionType",
```
SupressionType -> SuppressionType

roramc replied:
resolved in latest commit

nschonni commented on specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json:
```diff
@@ -0,0 +1,2062 @@
+        },
+        "recurrenceValues": {
+          "type": "array",
+          "description": "Specifies the values for reccurrence pattern",
```
reccurrence -> recurrence

AutorestCI commented:
# Automation for azure-sdk-for-node

AutorestCI commented:
# Automation for azure-sdk-for-java

roramcmsft committed:
Code review spelling and JSON fix

Spelling correction and JSON update for delete example

roramcmsft committed:
Update AlertsManagement.json

fixed semantic error - added recurrenceType in suppression config

praries880 commented:
@roramc The [name](https://github.com/Azure/azure-rest-api-specs/pull/4955/files#diff-5f2fc3c12561cd22b023ac3b6f46697b) contains "privatepreview".. if this is indeed a private preview that why not use the [private repo](https://github.com/Azure/azure-rest-api-specs-pr) to drive this?

praries880 commented on specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json:
```diff
@@ -0,0 +1,2062 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-05-05-preview",
```
is this expected? should the version be `2018-11-02-preview`?

roramc replied:
fixed in latest commit. Also, this preview version is just for the new resource type. The old resource types continue with the old version. So, should the old resource type definitions be removed from this .json file?

praries880 commented on specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json:
```diff
@@ -0,0 +1,2062 @@
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AlertsManagement/actionRules": {
+      "get": {
+        "summary": "Get all action rule in a given subscription",
+        "description": "List all action rules iof the subscription and given input filters",
```
typo...

```suggestion
        "description": "List all action rules of the subscription and given input filters",
```

roramc replied:
fixed in latest commit.

roramcmsft committed:
Update AlertsManagement.json

code review comments

praries880 commented:
@roramc waiting on ARM review before I can approve the PR.

ravbhatnagar commented on specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json:
```diff
@@ -0,0 +1,2062 @@
+          "description": "Last modified by user name.",
+          "readOnly": true
+        },
+        "enabled": {
```
string enums generally work better longer term. Specially when you want to expand states the property can take.

roramc replied:
Resolved in latest commit

ravbhatnagar commented on specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json:
```diff
@@ -0,0 +1,2062 @@
+          "description": "Last updated time of action rule. Date-Time in ISO-8601 format.",
+          "readOnly": true
+        },
+        "createdUserName": {
```
Not sure if this kind of PII is ok to have on the resource response. I will double check on my side.

roramc replied:
Ideally we would want to show who created/updated the given resource. Aren't other RPs doing the same? Do let us know if there is any concern with this.

ravbhatnagar reviewed (commented):
Only reviewed actionRule APIs since that seemed the new type that was added. Next time, please remember to make the first commit which is a copy of existing version. The new changes should come in the next commit onwards. It will make reviewing easier.

roramc commented on specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2018-11-02-privatepreview/AlertsManagement.json:
```diff
@@ -0,0 +1,2062 @@
+          "type": "string",
+          "description": "Description of action rule"
+        },
+        "resourceGroup": {
```
The same object gets returned even when we make a GET call to list all action rules for a given subscription. At that point user would want to know what resource group this action rule is stored in. So we added this in the properties explicitly. If  conveying it through Id is enough, we could remove this property as well. Not sure what is the convention.

roramcmsft committed:
Changed enabled flag to status enum

Changed enabled flag to status enum

roramcmsft committed:
added location as mandatory parameter

added location as mandatory parameter

praries880 commented:
@ravbhatnagar do the latest changess from @roramc  address all the issues you raised?

ravbhatnagar commented:
awaiting confirmation from service team. We recommended service team check with privacy since it contains potentially PII information.

roramcmsft committed:
changed "createdDateTime"/"createdUserName" to "createdAt"/"createdBy"

changed "createdDateTime"/"createdUserName" to "createdAt"/"createdBy"
changed "lastModifiedDateTime"/"lastModifiedUserName" to "lastModifiedAt"/"lastModifiedBy"
Updated conditions to take in operator

KrisBash reviewed (approved):
Thanks for the changes. Signing off

roramcmsft committed:
new line at end of file

new line at end of file

roramcmsft committed:
new line at end of file

new line at end of file
